lua/rocks/init.lua

```diff
@@ -9,21 +9,18 @@
 -- Homepage:   https://github.com/nvim-neorocks/rocks.nvim
 -- Maintainer: NTBBloodbath <bloodbathalchemist@protonmail.com>
 --
--------------------------------------------------------------------------------
+---@brief [[
 --
---- Commentary:
+-- rocks.nvim main module
 --
---
---
--------------------------------------------------------------------------------
---
---- Code:
+---@brief ]]
 
 local rocks = {}
 
 local setup = require("rocks.setup")
 local config = require("rocks.config")
 
+---@param opts RocksOptions
 function rocks.setup(opts)
     assert(vim.version() >= vim.version.parse("0.10.0-dev"), "rocks.nvim requires Neovim 0.10.0 or later!")
```

Comment:
Do we want to get rid of `setup`?

I would suggest to keep the `config` module and add the following:



```lua
--- @type RocksOptions
vim.g.rocks_nvim = vim.g.rocks_nvim -- for lua-language-server

---@type RocksConfig
local default_config = {
    ---@diagnostic disable-next-line: param-type-mismatch
    rocks_path = vim.fs.joinpath(vim.fn.stdpath("data"), "rocks"),
    ---@diagnostic disable-next-line: param-type-mismatch
    config_path = vim.fs.joinpath(vim.fn.stdpath("config"), "rocks.toml"),
}

local opts = validate(vim.g.rocks_nvim or {}) -- validate not implemented yet
---@type RocksConfig
local config = vim.tbl_deep_extend("force", config, opts or {})
return config
```

And to move the code in the `setup` module to the `plugin` directory.